e2e/e2e-windows-scenario.sh

```diff
@@ -0,0 +1,179 @@
+
+VMSS_INSTANCE_NAME=$(az vmss list-instances \
+                    -n ${DEPLOYMENT_VMSS_NAME} \
+                    -g $MC_RESOURCE_GROUP_NAME \
+                    -ojson | \
+                    jq -r '.[].osProfile.computerName')
+export DEPLOYMENT_VMSS_NAME
+export VMSS_INSTANCE_NAME
+
+VMSS_INSTANCE_ID=$(az vmss list-instances \
+                    -n ${DEPLOYMENT_VMSS_NAME} \
+                    -g $MC_RESOURCE_GROUP_NAME \
+                    -ojson | \
+                    jq -r '.[].instanceId'
+                )
+export VMSS_INSTANCE_ID
+
+# FAILED=0
+# # Check if the node joined the cluster
+# if [[ "$retval" != "0" ]]; then
+#     err "cse failed to apply"
+#     debug
+#     tail -n 50 $SCENARIO_NAME-logs/cluster-provision.log || true
+#     exit 1
+# fi
+
+# Sleep to let the automatic upgrade of the VM finish
+waitForNodeStartTime=$(date +%s)
+for i in $(seq 1 10); do
+    set +e
+    kubectl get nodes | grep $VMSS_INSTANCE_NAME
+    retval=$?
+    # pipefail interferes with conditional.
+    # shellcheck disable=SC2143
+    if [ -z "$(kubectl get nodes | grep $VMSS_INSTANCE_NAME | grep -v "NotReady")" ]; then
+        log "retrying attempt $i"
+        sleep 10
+        continue
+    fi
+    break;
+done
+waitForNodeEndTime=$(date +%s)
+log "Waited $((waitForNodeEndTime-waitForNodeStartTime)) seconds for node to join"
+
+FAILED=0
+# Check if the node joined the cluster
+if [ "$retval" -eq 0 ]; then
+    ok "Test succeeded, node joined the cluster"
+    kubectl get nodes -o wide | grep $VMSS_INSTANCE_NAME
+else
+    err "Node did not join cluster"
+    FAILED=1
+fi
+
+# Run a windows servercore pod on the node to check if pod runs
+POD_NAME=$(mktemp -u podName-XXXXXXX | tr '[:upper:]' '[:lower:]')
+export POD_NAME
+envsubst < pod-windows-template.yaml > pod-windows.yaml
+sleep 5
+kubectl apply -f pod-windows.yaml
+
+# Sleep to let Pod Status=Running
+waitForPodStartTime=$(date +%s)
+for i in $(seq 1 20); do
+    set +e
+    kubectl get pods -o wide | grep $POD_NAME
+    kubectl get pods -o wide | grep $POD_NAME | grep 'Running'
+    retval=$?
+    set -e
+    if [ "$retval" -ne 0 ]; then
+        log "retrying attempt $i"
+        sleep 15
+        continue
+    fi
+    break;
+done
+waitForPodEndTime=$(date +%s)
+log "Waited $((waitForPodEndTime-waitForPodStartTime)) seconds for pod to come up"
+
+if [ "$retval" -eq 0 ]; then
+    ok "Pod ran successfully"
+    # debug
+    mkdir -p $SCENARIO_NAME-logs
+    kubectl cp $POD_NAME:AzureData/CustomDataSetupScript.log $SCENARIO_NAME-logs/CustomDataSetupScript.log
+    kubectl cp $POD_NAME:AzureData/CustomDataSetupScript.ps1 $SCENARIO_NAME-logs/CustomDataSetupScript.ps1  
+else
+    err "Pod pending/not running"
+    kubectl get pods -o wide | grep $POD_NAME
+    kubectl describe pod $POD_NAME
+    exit 1
+fi
+
+if [ "$FAILED" == "1" ] || [ "$retval" -eq 1 ]; then
+    log "Reserve vmss and node for failed pipeline"
+else
+    waitForDeleteStartTime=$(date +%s)
```

Comment:
Can we not delete the node and vmss but delete the resource group directly without waiting to save time?